### PR TITLE
Allow the ability to specify a custom jshintrc file

### PR DESCRIPTION
**Pull request by jhalickman**

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -28,6 +28,11 @@ module.exports = {
       title: 'Lint JSX Files',
       type: 'boolean',
       default: false
+    },
+    jshintFilePath: {
```
Comment (Arcanemagus) [inline lib/main.js]: This is a name, not a full path. The variable name should reflect this.

File: lib/main.js [hunk 2]
```diff
@@ -93,14 +102,18 @@ module.exports = {
...
+        if (configFile) {
+          parameters.push('--config', configFile)
+        } 
+        
         if (this.disableWhenNoJshintrcFileInPath) {
-          configFile = await Helpers.findCachedAsync(Path.dirname(filePath), '.jshintrc')
-          if (configFile) {
-            parameters.push('--config', configFile)
-          } else {
+          if (!configFile) {
```
Comment (Arcanemagus) [inline lib/main.js]: Combine this condition as an `&&` in the above `if`.

**Comment by jhalickman** [comment]
Thanks, addressed the 2 comments.

File: lib/main.js [hunk 3]
```diff
@@ -93,14 +102,14 @@ module.exports = {
...
         let configFile = null
-        if (this.disableWhenNoJshintrcFileInPath) {
-          configFile = await Helpers.findCachedAsync(Path.dirname(filePath), '.jshintrc')
-          if (configFile) {
-            parameters.push('--config', configFile)
-          } else {
-            return results
-          }
+
+        configFile = await Helpers.findCachedAsync(Path.dirname(filePath), this.jshintFileName)
```
Comment (Arcanemagus) [inline lib/main.js]: Might as well combine this with the definition of the variable right above it.

**Comment by Arcanemagus** [comment]
One last comment and I'd say this is good :wink:.

**Comment by jhalickman** [comment]
No problem!

**Comment by Arcanemagus** [comment]
Thanks!

**Comment by Arcanemagus** [comment]
Published in v2.1.0 :tada:

**Pull request merged**